source/OptimaJet.DWKit.StarterApplication/Services/SendNotificationService.cs

```diff
@@ -0,0 +1,120 @@
+        public IJobRepository<UserRole> UserRolesRepository { get; }
+        public IJobRepository<Notification> NotificationRepository { get; }
+
+        public SendNotificationService(
+            ITranslator translator,
+            IJobRepository<Email> emailRepository,
+            IJobRepository<UserRole> userRolesRepository,
+            IJobRepository<Notification> notificationRepository
+        )
+        {
+            Translator = translator;
+            EmailRepository = emailRepository;
+            UserRolesRepository = userRolesRepository;
+            NotificationRepository = notificationRepository;
+
+        }
+        public async Task SendNotificationToOrgAdminsAsync(Organization organization, String message, dynamic subs)
+        {
+            var orgAdmins = UserRolesRepository.Get()
+                .Include(ur => ur.User)
+                .Include(ur => ur.Role)
+                .Where(ur => ur.OrganizationId == organization.Id && ur.Role.RoleName == RoleName.OrganizationAdmin)
+                .ToList();
+            foreach (UserRole orgAdmin in orgAdmins)
+            {
+                await SendNotificationToUserAsync(orgAdmin.User, message, subs);
+            }
+        }
+        public async Task SendNotificationToUserAsync(User user, String message, dynamic subs)
+        {
+            var notification = new Notification
+            {
+                MessageId = message,
+                UserId = user.Id,
+                MessageSubstitutions = subs
+            };
+            var updatedNotification = await NotificationRepository.CreateAsync(notification);
+            await HubContext.Clients.User(user.ExternalId).SendAsync("Notification", updatedNotification.Id);
+        }
+        public void NotificationEmailMonitor()
+        {
+            // Get limits from environment
+            int sendNotificationEmailMinutes = GetIntVarOrDefault("SEND_NOTIFICATION_EMAIL_MINUTES", 60);
+            int dontSendNotificationEmailMinutes = GetIntVarOrDefault("DONT_SEND_NOTIFICATION_EMAIL_MINUTES", 180);
+            var notifications = NotificationRepository.Get()
+                                                    .Where(n => n.DateEmailSent == null && n.DateRead == null)
+                                                    .Include(n => n.User)
+                                                    .ToList();
+            foreach (var notification in notifications)
+            {
+                if (ShouldSendEmail(notification, sendNotificationEmailMinutes, dontSendNotificationEmailMinutes))
+                {
+                    SendEmailAsync(notification).Wait();
+                }
+            }
+        }
+        protected bool ShouldSendEmail(Notification notification, int beginMinutes, int endMinutes)
+        {
+            var sendEmail = false;
+            var now = DateTime.UtcNow;
+            var dateCreated = notification.DateCreated ?? now;
```

Comment:
I guess I put it in because the test case hit that and blew up. Seemed like a valid check just in case.  I can manually set it for that case in the test and remove this.